packages/stryker/test/unit/TempFolderSpec.ts

```diff
@@ -0,0 +1,84 @@
+import { TempFolder } from '../../src/utils/TempFolder';
+
+import * as sinon from 'sinon';
+import { expect } from 'chai';
+import * as mkdirp from 'mkdirp';
+import * as fs from 'mz/fs';
+
+describe.only('TempFolder', () => {
+  let sandbox: sinon.SinonSandbox;
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create();
+  });
+  afterEach(() => sandbox.restore());
+
+  it('TempFolder is presented', () => {
+    expect(TempFolder).not.to.be.undefined;
+
+    const tempFolderInstance = TempFolder.instance();
+
+    expect(tempFolderInstance.createRandomFolder).to.exist;
+    expect(tempFolderInstance.writeFile).to.exist;
+    expect(tempFolderInstance.copyFile).to.exist;
+    expect(tempFolderInstance.clean).to.exist;
+    expect(tempFolderInstance.random).to.exist;
+  });
+
+  describe('createRandomFolder', () => {
+    it('should create dir with correct path', () => {
+      const mockCwd = process.cwd() + '/some/dir';
+      sandbox.stub(mkdirp, 'sync');
+      sandbox.stub(process, 'cwd').returns(mockCwd);
+      sandbox.stub(TempFolder.instance(), 'random').returns('rand');
```

Comment:
I think it's better to move this stubbing to the top before each. Putting it right here clutters the tests a bit. You can still do the `cwdStub.returns(mockCwd)` and `randomStub.returns('rand')` here.

Reply:
Fixed